layer/event_logging.h

```diff
@@ -84,6 +87,43 @@ class AttributeImpl : public Attribute {
   T value_;
 };
 
+// The base class for an attribute that keeps the duration information. To take
+// the time unit into account, the subclasses must implement the `GetValue()`
+// method such that it converts the underlying `duration` value to a specific
+// unit(nanoseconds, miliseconds, seconds, etc).
+class DurationAttr : public Attribute {
+ public:
+  static constexpr ValueType id_ = ValueType::kDuration;
+
+  virtual const int64_t &GetValue() const = 0;
+
+ protected:
+  DurationAttr(const char *name, const DurationClock::duration &value)
+      : Attribute(name, ValueType::kDuration), value_(value) {}
+
+  virtual int64_t &GetValue() = 0;
+
+  const DurationClock::duration &GetDuration() { return value_; }
+
+ private:
+  DurationClock::duration value_;
+};
+
+// An implementation of `DurationAttr` for `nanoseconds`.
+class DurationNsAttr : public DurationAttr {
```

Comment:
I think DurationNs should the only DurationAttribute. We can then add more getter functions if we need less than nanosecond precision.